edu/tamu/csce434/Compiler.java

```diff
@@ -1231,17 +1256,18 @@ private void STATSEQUENCE() {
 			
 			// does stuff (Actually looks through the tree and finds all parents looking at this removed child
 			// we can then redirect them back into the child-less parent list)
-			for (int i=0; i<BlockChain.size(); i++) {
-				if(BlockChain.get(i).childrenIndexes.contains(BlockChain.size())) {
-					ChildlessParents.add(BlockChain.get(i).BlockNumber);
-					for (int j=0; j<BlockChain.get(i).childrenIndexes.size(); j++) {
-						if (BlockChain.get(i).childrenIndexes.get(j) == BlockChain.size()) {
-							BlockChain.get(i).childrenIndexes.remove(j);
-							continue;
-						}
-					}
-				}
-			}
+			removeEmptyBlocks();
```

Comment:
@philipely Did we want the conditional check to also be part of the helper function? Or just the removing of empty blocks itself?

Reply:
Updated this